Assets/Scripts/Model/Core.cs

```diff
@@ -169,5 +170,29 @@ internal Player GetPlayer(int x, int y)
             }
             return null;
         }
+
+        private TurnMode lastturnmode;
+        public TurnMode LastTurnMode {
+            get { return lastturnmode; }
+            set {
+                if (value == lastturnmode) return;
+                lastturnmode = value;
+                Debug.Log($"LastTurnMode: {lastturnmode.Name}");
+            }
+        }
+        private TurnMode turnmode;
+        public TurnMode TurnMode {
```

Comment:
A property setter like this shouldn't have this much side-effects. Convert to a simpler variant:
`public TurnMode TurnMode { get; private set; }`
and implement a `SetTurnMode(TurnMode mode)` method that manages the last turn mode. Note that if you insist on having debug output for lastturnmode, this is where it should be rather than in the property.